### ISSUE TITLE
Update wgpu to v23

### ISSUE THREAD

**Issue by torsteingrindvik** [tracel-ai/cubecl#343]
Currently if you try to use `comptime! { .. }` to build `Array`s in cube kernels accessing them works fine for CUDA.

Using WGPU + WGSL doesn't work. This was recently fixed, the v23 changelog shows:

(See https://github.com/gfx-rs/wgpu/blob/trunk/CHANGELOG.md#contributor-spotlight-sagudev)
![image](https://github.com/user-attachments/assets/a8ebf7f7-18ec-4c92-8043-dc07d1825f21)

On `main` this currently happens on WGPU + WGSL:

```
wgpu error: Validation Error

Caused by:
  In Device::create_shader_module

Shader validation error:
   ┌─ :62:11
   │
62 │ let _24 = arrays_0[_23];
   │           ^^^^^^^^^^^^^ naga::Expression [84]


      Entry point foo_f32 at Compute is invalid
        Expression [84] is invalid
          The expression [83] may only be indexed by a constant
```

on WGPU + SPIRV:

```
thread 'main' panicked at $PATH/cubecl/crates/cubecl-spirv/src/variable.rs:399:49:
index out of bounds: the len is 0 but the index is 0
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

A "quick and dirty" update to v23 locally fixed the WGPU + WGSL issue for me.

For a proper update a few things must be cleared up:

- `wgpu::Id` isn't a thing anymore, how should `WgpuDevice::Existing` work now?
- This line has changed in v23 to accept a callback fn: https://github.com/tracel-ai/cubecl/blob/main/crates/cubecl-wgpu/src/compiler/spirv.rs#L283 , should it be used or should just `None` be set?

**Comment by AsherJingkongChen** [comment]
Hi, I am working on #211. You can write comments there.

**Comment by torsteingrindvik** [comment]
Fixed in #211 

Thanks @AsherJingkongChen
